core/client/fs/src/main/java/alluxio/client/file/cache/store/PageStoreDir.java

```diff
@@ -0,0 +1,168 @@
+import alluxio.client.file.cache.evictor.CacheEvictor;
+import alluxio.conf.AlluxioConfiguration;
+
+import com.google.common.collect.ImmutableList;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * Directory of page store.
+ */
+public interface PageStoreDir {
+  /**
+   * Create a list of PageStoreDir based on the configuration.
+   * @param conf AlluxioConfiguration
+   * @return A list of LocalCacheDir
+   * @throws IOException
+   */
+  static List<PageStoreDir> createPageStoreDirs(AlluxioConfiguration conf)
+      throws IOException {
+    return PageStoreOptions.create(conf).stream().map(options -> createPageStoreDir(conf, options))
+        .collect(ImmutableList.toImmutableList());
+  }
+
+  /**
+   * Create an instance of PageStoreDir.
+   *
+   * @param conf
+   * @param pageStoreOptions
+   * @return PageStoreDir
+   */
+  static PageStoreDir createPageStoreDir(AlluxioConfiguration conf,
+                                         PageStoreOptions pageStoreOptions) {
+    switch (pageStoreOptions.getType()) {
+      case LOCAL:
+        checkState(pageStoreOptions instanceof LocalPageStoreOptions);
+        return new LocalPageStoreDir(
+            (LocalPageStoreOptions) pageStoreOptions,
+            PageStore.openOrCreatePageStore(pageStoreOptions),
+            CacheEvictor.create(conf)
+        );
+      case ROCKS:
+        return new RocksPageStoreDir(
+            pageStoreOptions,
+            PageStore.openOrCreatePageStore(pageStoreOptions),
+            CacheEvictor.create(conf)
+        );
+      case MEM:
+        return new MemoryPageStoreDir(
+            pageStoreOptions,
+            (MemoryPageStore) PageStore.openOrCreatePageStore(pageStoreOptions),
+            CacheEvictor.create(conf)
+        );
+      default:
+        throw new IllegalArgumentException(String.format("Unrecognized store type %s",
+            pageStoreOptions.getType().name()));
+    }
+  }
+
+  /**
+   * @param fileBuckets number of buckets
+   * @param fileId file id
+   * @return file bucket
+   */
+  static String getFileBucket(int fileBuckets, String fileId) {
+    return Integer.toString(Math.floorMod(fileId.hashCode(), fileBuckets));
+  }
+
+  /**
+   * @return root path
+   */
+  Path getRootPath();
+
+  /**
+   * Getter for pageStore.
+   * @return pageStore
+   */
+  PageStore getPageStore();
+
+  /**
+   * Getter for capacity.
+   * @return capacity
+   */
+  long getCapacity();
+
+  /**
+   * Reset page store.
+   */
+  void reset();
+
+  /**
+   * Traverse the pages under this dir.
+   * @param pageInfoConsumer
+   * @throws IOException
+   */
+  void restorePages(Consumer<PageInfo> pageInfoConsumer) throws IOException;
+
+  /**
+   * @return cached bytes in this directory
+   */
+  long getCachedBytes();
+
+  /**
+   * @param pageInfo
+   * @return if the page added successfully
+   */
+  boolean putPageToDir(PageInfo pageInfo);
```

Comment:
Feels that the `toDir` part is not necessary in the APIs. Also, what's the difference between `putXXX` and `addXXX`?

Reply:
renamed to putPage, putTempFile to align with the LocalCacheManager.  I don't think there is any difference because the cache is immutable, so put and add are identical.